src/FSharp.Data.GraphQL.Server.AppInfrastructure/Serialization/GraphQLQueryDecoding.fs

```diff
@@ -0,0 +1,76 @@
+namespace FSharp.Data.GraphQL.Server.AppInfrastructure
+
+module GraphQLQueryDecoding =
+  open FSharp.Data.GraphQL
+  open Rop
+  open System
+  open System.Text.Json
+  open System.Text.Json.Serialization
+
+  let private resolveVariables (serializerOptions : JsonSerializerOptions) (expectedVariables : Types.VarDef list) (variableValuesObj : JsonDocument) =
```

Comment:
Same here, it is already automated inside `IGQLError` branch code

Reply:
Good to know. Looking forward to the IGQLError branch being merged.